### ISSUE TITLE
Auto append rbe gn config options based on gclient_variables

### ISSUE THREAD

**Issue by keyonghan** [flutter/flutter#144269]
Existing targets running based on `rbe` need to specify:
```
"gclient_variables": {
                "use_rbe": true
            },
```
and 
```
"gn": [
                "--rbe",
                "--no-goma"
          ]
```

Tons of duplication are in each subbuild's configs. We can have recipes to auto append gn configs based on the `use_rbe` gclient variable. The assumption is: `"use_rbe": true` is quivalent to `"--rbe" && "--no-goma"`

**Comment by zanderso** [comment]
So, my preference would actually be to reduce the amount of implicit configuration that happens in recipes, instead explicitly giving all configuration in the build config files that live in the engine repo.

In particular, RBE will require additional tooling in the local development workflow anyway, so that's the right place to streamline rather than adding implicit logic that lives outside the flutter/engine repo.

tl;dr: let's please not do this.

**Comment by keyonghan** [comment]
This is an up-to-team preference. Closing with the above comment.
